python/ray/autoscaler/autoscaler.py

```diff
@@ -824,3 +854,22 @@ def add_hash_of_file(fpath):
         _hash_cache[conf_str] = hasher.hexdigest()
 
     return _hash_cache[conf_str]
+
+def request_cores(redis_address, cores, redis_password=None):
```

Comment:
Can we make this a bit more general: `ray.autoscaler.request_resources(num_cpus=None, num_gpus=None)`?

Reply:
Can do although we haven't implemented the GPU bit yet - are you happy for us to just raise a NotImplementedError for now if someone requests those?

Reply:
That sounds good.

Reply:
Done!